azext_iot/tests/iothub/core/test_iothub_storage_int.py

```diff
@@ -321,49 +322,58 @@ def test_user_identity_storage(self):
         identity_principal = hub_identity["userAssignedIdentities"][identity_id]["principalId"]
         assert identity_principal == user_identity["principalId"]
 
-        self.assign_storage_role_if_needed(identity_principal)
+        attempts = 0
+        user_identity_setup_completed = False
+        while not user_identity_setup_completed:
+            try:
+                self.assign_storage_role_if_needed(identity_principal)
 
-        # user identity assignment takes longer than expected
-        sleep(30)
+                # user identity assignment takes longer than expected
+                sleep(30)
```

Comment:
Since `assign_storage_role_if_needed()` waits for role assignment propagation is this `sleep` needed?

Reply:
We can get rid of it, was done earlier for safety